src/ConfigurationHelper.php

```diff
@@ -0,0 +1,539 @@
+      '%content-type' => $content_type,
+      '%block-id' => $block_id,
+    ]);
+
+    return TRUE;
+  }
+
+  /**
+   * Add entity bundle to index datasource.
+   *
+   * @param \Drupal\search_api\IndexInterface $index
+   *   The index to add bundle to.
+   * @param string $entity_type_id
+   *   Entity type ID.
+   * @param string $entity_bundle
+   *   The bundle ID.
+   */
+  protected function indexAddBundle(IndexInterface $index, string $entity_type_id, string $entity_bundle): void {
+    $datasource = $this->indexGetDatasource($index, $entity_type_id);
+    if (!$datasource) {
+      $this->logger->error('Failed to update the directories search index with new bundle');
+      return;
+    }
+
+    $configuration = $datasource->getConfiguration();
+    $configuration['bundles']['default'] = FALSE;
+    if (!in_array($entity_bundle, $configuration['bundles']['selected'])) {
+      $configuration['bundles']['selected'][] = $entity_bundle;
+    }
+    $datasource->setConfiguration($configuration);
+  }
+
+  /**
+   * Remove entity bundle to index datasource.
+   *
+   * @param \Drupal\search_api\IndexInterface $index
+   *   The index to remove bundle from.
+   * @param string $entity_type_id
+   *   Entity type ID.
+   * @param string $entity_bundle
+   *   The bundle ID.
+   */
+  protected function indexRemoveBundle(IndexInterface $index, string $entity_type_id, string $entity_bundle): void {
+    $datasource = $this->indexGetDatasource($index, $entity_type_id);
+    if (!$datasource) {
+      $this->logger->error('Failed to update the directories search index with new bundle');
+      return;
+    }
+
+    $configuration = $datasource->getConfiguration();
+    $configuration['bundles']['default'] = FALSE;
+    if (($key = array_search($entity_bundle, $configuration['bundles']['selected'])) !== FALSE) {
+      unset($configuration['bundles']['selected'][$key]);
+    }
+    $datasource->setConfiguration($configuration);
+  }
+
+  /**
+   * Set entity bundle default view mode in view.
+   *
+   * @param \Drupal\views\ViewEntityInterface $view
+   *   The view to update.
+   * @param string $entity_type_id
+   *   Entity type ID.
+   * @param string $entity_bundle
+   *   The bundle ID.
+   */
+  protected function viewSetViewMode(ViewEntityInterface $view, string $entity_type_id, string $entity_bundle): void {
+    // Also set the default view mode for the directory view listing.
+    $display = $view->get('display');
+    if (isset($display['node_embed']['display_options']['row'])) {
+      $display['node_embed']['display_options']['row']['options']['view_modes']['entity:' . $entity_type_id][$entity_bundle] = 'teaser';
+    }
+    elseif (isset($display['default']['display_options']['row'])) {
+      $display['default']['display_options']['row']['options']['view_modes']['entity:' . $entity_type_id][$entity_bundle] = 'teaser';
+    }
+    $view->set('display', $display);
```

Comment:
The `$view` has not been saved here.  Is that okay?

Reply:
$view is being passed into the method; as is $index in the other similar methods, the caller is saving them. In this case it could be as well in the method, but it's consistent, in the others it enables running multiple changes in different methods and saving once.